aten/src/ATen/native/cuda/CusparseLtKernels.cpp

```diff
@@ -0,0 +1,407 @@
+/*
+The following source file implements a sparse linear operator using cusparseLt
+*/
+#include <torch/custom_class.h>
```

Comment:
We can't have dependencies on `torch` within aten.

Reply:
That means we'd need to move this construct, if it's going to use custom_class, into the torch folder. We can land native functions for the CUDA dispatch key for some of the core routines.

Reply:
Makes sense. Right now we need custom class to keep track of some state like the alg plan and weight matrix descriptor, but I think I might be able to refactor.